subprojects/docs/src/docs/release/notes.md

```diff
@@ -484,6 +493,11 @@ Please ensure that all builds being executed with Gradle TestKit define `setting
 Adding `-sourcepath` or `--source-path` to the `CompileOptions.compilerArgs` list is now prohibited. 
 The source path for a `JavaCompile` task should be set via the `CompileOptions.sourcePath` property.
 
+### Cannot specify `--processor-path` directly as a Java compiler arg
+
+Adding `-processorpath` or `--processor-path` to the `CompileOptions.compilerArgs` list is now prohibited.
+The annotation processor path for a `JavaCompile` task should be set via the `CompileOptions.annotationProcessorPath` property.
```

Comment:
Should we also say something here that the `JavaCompile` tasks created by Gradle plugins will automatically have this wired into a configuration?  It's much better to have someone add dependencies to the configuration than to create their own or set `annotationProcessorPath` to a raw `files(...)` instead.